src/wire/libevent_server.cpp

```diff
@@ -126,14 +151,27 @@ LibeventServer::LibeventServer() {
 
     LOG_INFO("Listening on port %lu", port_);
     event_base_dispatch(base);
+    LibeventServer::GetConn(listen_fd)->CloseSocket();
+    event_free(LibeventServer::GetConn(listen_fd)->event);
     event_free(evstop);
+    event_free(ev_timeout);
     event_base_free(base);
+    static_cast<LibeventMasterThread *>(master_thread.get())->CloseConnection();
+    LOG_INFO("Server Closed");
   }
 
   // This socket family code is not implemented yet
   else {
     throw ConnectionException("Unsupported socket family");
   }
 }
+
+void LibeventServer::CloseServer() {
```

Comment:
In CloseServer(), I was wondering whether it's possible to free some resource used by the "global_socket_list" since the memory for each LibeventSocket is dynamically allocated using new. Maybe the number of "delete" should be the same as the number of "new".